fl4health/model_bases/masked_layers/masked_conv.py

```diff
@@ -441,23 +357,425 @@ def from_pretrained(cls, conv_module: nn.Conv3d) -> "MaskedConv3d":
+    ) -> None:
+        """
+        Implementation of masked ConvTranspose1d layers.
+
+        Like regular ConvTranspose1d layers (i.e., nn.ConvTranspose1d module),
+        a masked transpose convolutional layer has a weight (i.e., convolutional filter)
+        and a (optional) bias.
+        However, the weight and the bias do not receive gradient in back propagation.
+        Instead, two score tensors - one for the weight and another for the bias - are maintained.
+        In the forward pass, the score tensors are transformed by the Sigmoid function into probability scores,
+        which are then used to produce binary masks via bernoulli sampling.
+        Finally, the binary masks are applied to the weight and the bias. During training,
+        gradients with respect to the score tensors are computed and used to update the score tensors.
+
+        Note: the scores are not assumed to be bounded between 0 and 1.
+
+
+        Args:
+            in_channels (int): Number of channels in the input image
+            out_channels (int): Number of channels produced by the convolution
+            kernel_size (int or tuple): Size of the convolving kernel
+            stride (int or tuple, optional): Stride of the convolution. Default: 1
+            padding (int or tuple, optional): ``dilation * (kernel_size - 1) - padding`` zero-padding
+                will be added to both sides of the input. Default: 0
+            output_padding (int or tuple, optional): Additional size added to one side
+                of the output shape. Default: 0
+            groups (int, optional): Number of blocked connections from input channels to output channels. Default: 1
+            bias (bool, optional): If ``True``, adds a learnable bias to the output. Default: ``True``
+            dilation (int or tuple, optional): Spacing between kernel elements. Default: 1
+
+
+        Attributes:
+            weight (Tensor): weights of the module.
+            bias (Tensor):   bias of the module.
+            weight_score: learnable scores for the weights. Has the same shape as weight.
+            bias_score: learnable scores for the bias. Has the same shape as bias.
+        """
+        super().__init__(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride,
+            padding,
+            output_padding,
+            groups,
+            bias,
+            dilation,
+            padding_mode,
+            device,
+            dtype,
+        )
+        self.weight.requires_grad = False
+        self.weight_scores = Parameter(torch.randn_like(self.weight), requires_grad=True)
+        if bias:
+            assert self.bias is not None
+            self.bias.requires_grad = False
+            self.bias_scores = Parameter(torch.randn_like(self.bias), requires_grad=True)
+        else:
+            self.register_parameter("bias_scores", None)
+        self.reset_parameters()
```

Comment:
This function is used to initialize the weight and bias tensor following the Kaiming uniform distribution since both the weight and bias are initialized via `torch.empty` in the parent class. But I now noticed `self.reset_parameters()` is already being called in the parent class so I've removed them here.